injection/sharedmain/main.go

```diff
@@ -58,6 +57,7 @@ import (
 
 func init() {
 	maxprocs.Set()
+	flag.StringVar(&disabledControllers, "disable-controllers", "", "Comma-separated list of disabled controllers.")
```

Comment:
Can we define this inside the `MainNamed` function?

If a controller is not using MainNamed and there is a `disable-controllers` flag defined we would get an error of undefined flag.
With a global flag, we silently define the flag for every importer.